front/lib/upsert_queue.ts

```diff
@@ -84,6 +84,15 @@ export async function enqueueUpsertDocument({
     "[UpsertQueue] Enqueueing document"
   );
 
+  if (
+    upsertDocument.parentId &&
+    upsertDocument.parents?.[1] !== upsertDocument.parentId
+  ) {
+    throw new Error(
```

Comment:
Can we also have a validation in the handler with a proper 400? 🙏

Reply:
Done